tests/profiling_v2/collector/test_threading.py

```diff
@@ -807,3 +807,85 @@ def test_global_locks(self):
                 ),
             ],
         )
+
+    def test_lock_acquire_fail(self):
+        collector = collector_threading.ThreadingLockCollector(None, capture_pct=100, export_libdd_enabled=True)
+        with collector:
+            lock = threading.Lock()  # !CREATE! test_lock_acquire_fail
+            lock.acquire()  # !ACQUIRE! test_lock_acquire_fail
+
+            # Try to acquire the lock again, this should fail. To test this
+            # behavior, we set blocking=False so that the lock doesn't block
+            ret = lock.acquire(blocking=False)
+            assert not ret
+
+        ddup.upload()
+
+        # parse_profile raises an AssertionError if it doesn't have any sample
+
+        profile = pprof_utils.parse_profile(self.output_filename)
+
+        linenos = get_lock_linenos("test_lock_acquire_fail")
+        pprof_utils.assert_lock_events(
```

Comment:
I might be missing something, but does this cover what we want? It'll show the one acquire event for the acquire that succeeds, but should it also somehow assert that we see _only_ that one? Like, that we don't see a second event for the failed acquire?

Reply:
oh yeah, good catch. assert_lock_events only checks that the profile contains expected samples. Added an assert stmt to check that the profile only contains 1 acquire sample